pubsub/message.go

```diff
@@ -15,55 +15,50 @@
 package pubsub
 
 import (
+	"fmt"
 	"time"
 
+	internalpubsub "cloud.google.com/go/internal/pubsub"
 	"github.com/golang/protobuf/ptypes"
 	pb "google.golang.org/genproto/googleapis/pubsub/v1"
 )
 
 // Message represents a Pub/Sub message.
-type Message struct {
-	// ID identifies this message. This ID is assigned by the server and is
-	// populated for Messages obtained from a subscription.
-	//
-	// This field is read-only.
-	ID string
-
-	// Data is the actual data in the message.
-	Data []byte
-
-	// Attributes represents the key-value pairs the current message is
-	// labelled with.
-	Attributes map[string]string
-
-	// PublishTime is the time at which the message was published. This is
-	// populated by the server for Messages obtained from a subscription.
-	//
-	// This field is read-only.
-	PublishTime time.Time
-
-	// DeliveryAttempt is the number of times a message has been delivered.
-	// This is part of the dead lettering feature that forwards messages that
-	// fail to be processed (from nack/ack deadline timeout) to a dead letter topic.
-	// If dead lettering is enabled, this will be set on all attempts, starting
-	// with value 1. Otherwise, the value will be nil.
-	// This field is read-only.
-	DeliveryAttempt *int
-
-	// size is the approximate size of the message's data and attributes.
-	size int
-
-	// OrderingKey identifies related messages for which publish order should
-	// be respected. If empty string is used, message will be sent unordered.
-	OrderingKey string
-
-	// ackh handles Ack() or Nack().
-	ackh ackHandler
+type Message = internalpubsub.Message
```

Comment:
Only downside of this is that the godoc will now be a little less clear(indirect). Takes an extra click to see all of the exported fields. Seems like an okay tradeoff though. Example: [Api](https://pkg.go.dev/google.golang.org/genproto@v0.0.0-20201203001206-6486ece9c497/protobuf/api#Api)

Reply:
Agreed. In the latest commit for this PR, I elaborated on the comments for the aliases to save users having to click through in some cases.

I also changed the import to `ipubsub` (more for aesthetic reasons), but can change it back to `internalpubsub` if it's better to be clear about where the actual types are defined.